### PR TITLE
[DPE-1760] Juju 3

### PR DESCRIPTION
**Pull request by dragomirp**
## Issue
We should run integration tests against both juju 3 and juju 2

## Solution
* Add support for running integration test on both juju versions
* Add renovate black magic

### CODE REVIEW

File: renovate.json
Comment (dragomirp) [inline renovate.json:1]: Copypaste from the pgbouncer charm

File: .github/workflows/ci.yaml
Comment (dragomirp) [inline .github/workflows/ci.yaml:1]: Copypaste from the pgbouncer charm

File: tox.ini
Comment (dragomirp) [inline tox.ini:1]: Copypaste of the logic in pgbouncer.

File: tests/integration/ha_tests/scripts/deploy_chaos_mesh.sh [hunk 1]
```diff
@@ -12,10 +12,10 @@ fi
 
 deploy_chaos_mesh() {
 	echo "adding chaos-mesh helm repo"
-	sg microk8s -c "microk8s.helm3 repo add chaos-mesh https://charts.chaos-mesh.org"
+	sg snap_microk8s -c "microk8s.helm3 repo add chaos-mesh https://charts.chaos-mesh.org"
```
Comment (dragomirp) [inline tests/integration/ha_tests/scripts/deploy_chaos_mesh.sh:15]: The group is different for the strictly confined snap

File: pyproject.toml [hunk 2]
```diff
@@ -16,7 +16,7 @@ python = "^3.10.6"
 ops = "2.4.1"
 pgconnstr = "1.0.1"
 tenacity = "8.2.2"
-cryptography = "41.0.1"
+cryptography = "41.0.2"
```
Comment (dragomirp) [inline pyproject.toml:19]: Got errors locally with poetry, so bumped just in case.

File: tests/integration/new_relations/test_new_relations.py [hunk 3]
```diff
@@ -233,18 +234,20 @@ async def test_an_application_can_connect_to_multiple_aliased_database_clusters(
...
+                ALIASED_MULTIPLE_DATABASE_CLUSTERS_RELATION_NAME,
+                relation_alias="cluster1",
+            )
+            another_application_connection_string = await build_connection_string(
+                ops_test,
+                APPLICATION_APP_NAME,
+                ALIASED_MULTIPLE_DATABASE_CLUSTERS_RELATION_NAME,
+                relation_alias="cluster2",
+            )
     assert application_connection_string != another_application_connection_string
```
Comment (dragomirp) [inline tests/integration/new_relations/test_new_relations.py:251]: Frequently fails. Added retrying, but I'm not sure it helps.

File: src/relations/postgresql_provider.py [hunk 4]
```diff
@@ -119,7 +119,8 @@ def _on_relation_broken(self, event: RelationBrokenEvent) -> None:
         """Remove the user created for this relation."""
         # Check for some conditions before trying to access the PostgreSQL instance.
         if (
-            "cluster_initialised" not in self.charm._peers.data[self.charm.app]
+            not self.charm._peers
```
Comment (dragomirp) [inline src/relations/postgresql_provider.py:122]: When units departed, this check fails due to `_peers` being None, leaving the unit stuck in removing in Juju 3.

**Review by taurus-forever — APPROVED** [review]
Thank you! CC: @carlcsaposs-canonical please check it from your side too (re: mysql 3.1 migration tasks)!

**Review by marceloneppel — APPROVED** [review]

**Pull request merged**